docs/source/format/Columnar.rst

```diff
@@ -981,6 +981,11 @@ a ``RecordBatch`` it should be defined in a ``DictionaryBatch``. ::
     <RECORD BATCH n - 1>
     <EOS [optional]: 0xFFFFFFFF 0x00000000>
 
+.. note:: An edge-case for interleaved dictionary and record batches occurs
+   when the the record batches contain dictionary encoded arrays that are
+   completely null. In this case, the dictionary for the encoded column might
+   appear after the first record batch.
```

Comment:
Is this necessary? This may complicate implementations.

Reply:
i'm fine if we revise, the specification to require a dictionary for each ID in the schema even if it is empty.  Requiring all dictionaries at the beginning seemed like a change in specification which is why I proposed this.  Lets see if anyone else has any thoughts on it.

Reply:
I'm not especially concerned about this at least on the C++ side.

Reply:
The alternative to this is to require an empty dictionary batch to be sent. It is also difficult to know whether the dictionary _should_ be empty without checking the dictionary indices.

Reply:
Yes, I don't think either option is too bad we just need to decide whether an empty batch is required.  I think if require implementations to support dictionary delta encoding and dictionary resets I don't think it make it any easier on the consumer side to require a dictionary (on the producer side I can imagine most implementations might not bother checking).